tflearn/helpers/trainer.py

```diff
@@ -87,8 +87,12 @@ def __init__(self, train_ops, graph=None, clip_gradients=5.0,
             self.validate_trainop_names()
 
             self.global_loss = None
-            self.global_step = tf.Variable(0., name='Global_Step',
-                                           trainable=False)
+            if global_step:
+                self.global_step = global_step
+            else:
```

Comment:
I thought about that but putting everything in one line might hurt readability: after all, the definition of tf.Variable(...) is already taking two lines.